fastestimator/pipeline.py

```diff
@@ -519,3 +551,57 @@ def _pad_batch_collate(self, batch: List[MutableMapping[str, Any]]) -> Dict[str,
         """
         pad_batch(batch, self.pad_value)
         return default_collate(batch)
+
+
+class ExtendDatasetSampler(Sampler):
+    """Samples elements randomly, while ensuring change in size of dataset.
+    Arguments:
+        ds_len : Length of original dataset.
+        ds_expand_len : Length to which original dataset must be expanded to.
+        shuffle : Whether to use shuffling.
+    """
+    def __init__(self, ds_len, ds_expand_len, shuffle=True):
+
+        self.ds_len = ds_len
+        self.ds_expand_len = ds_expand_len
+        self.shuffle = shuffle
+
+        self._check_input()
+
+        self.indices = []
+        self.base = [[i for i in range(self.ds_len)] for _ in range(math.ceil(self.ds_expand_len / self.ds_len))]
+
+    def __len__(self):
+        return self.ds_expand_len
+
+    def _check_input(self) -> None:
+        """Verify that the given input values are valid.
+        Raises:
+            AssertionError: If any of the parameters are found to by unacceptable for a variety of reasons.
+        """
+        assert isinstance(self.ds_expand_len, int), "Only accept positive integer type as ds_expand_len"
+        assert self.ds_expand_len > 0, "Invalid ds_expand_len. Expand Length cannot be less than or equal to 0"
+        assert isinstance(self.ds_len, int), "Only accept positive integer type as ds_len"
+        assert self.ds_len > 0, "Invalid ds_len. Dataset Length cannot be less than or equal to 0"
+
+    def __iter__(self):
+
+        self.indices = []
+        base = self.base
+
+        if self.ds_expand_len == self.ds_len:
```

Comment:
this if condition isn't really necessary since that special case should work just fine inside the else block

Reply:
In else part, this slicing line might produce an empty array if both the lengths are equal. It might make the array empty and return nothing.

![image](https://user-images.githubusercontent.com/44550569/142955683-c5b18740-3ffa-404e-9c76-8a13e8463979.png)

Reply:
ah, true. it may be better to use an if ds_expand_len % self.ds_len != 0 for the slicing instead though. The advantage being that less code is replicated between branches in case we add more features in future